src/pkg/caendr/caendr/models/datastore/wormbase.py

```diff
@@ -0,0 +1,145 @@
+
+        # Filter out types other than str
+        elif not isinstance(value, str):
+            logger.warning(f'Cannot set WormBase Version with object of type {type(value)} (expected str)')
+            raise InternalError()
+
+        # Validate desired new value matches expected format
+        elif not WormbaseVersion.validate(value):
+            logger.warning(f'Invalid WormBase Version String: "{value}"')
+            raise InternalError()
+
+        # If all checks passed, use uppercase string value
+        else:
+            self._value = value.upper()
+
+
+    def __repr__(self):
+        return f'<WB Ver "{self._value}">'
+
+    def __str__(self):
+        return self._value
+
+    def __eq__(self, other):
+        return self._value == other._value
+
+    def __bool__(self):
+        return self._value is not None
+
+
+    @classmethod
+    def validate(cls, value):
+
+        # Return None as-is
+        if value is None:
+            return None
+
+        # Validate value field of WormbaseVersion class
+        elif isinstance(value, WormbaseVersion):
+            return WormbaseVersion.validate(value._value)
+
+        # Check string value against RegEx 'WS###', e.g. 'WS276'
+        return isinstance(value, str) and bool(re.match(r'^WS[0-9]+$', value, flags=re.IGNORECASE))
+
+
+
+class WormbaseProjectNumber():
+
+    def __init__(self, value):
+
+        if value is None:
+            self._value = None
+
+        # Copy value of another Wormbase Project Number object
+        elif isinstance(value, WormbaseProjectNumber):
+            self._value = value._value
+
+        # Filter out types other than str
+        elif not isinstance(value, str):
+            logger.warning(f'Cannot set WormBase Project Number with object of type {type(value)} (expected str)')
+            raise InternalError()
+
+        # Validate desired new value matches expected format
+        elif not WormbaseProjectNumber.validate(value):
+            logger.warning(f'Invalid WormBase Project Number String: "{value}"')
+            raise InternalError()
+
+        # If all checks passed, use uppercase string value
+        else:
+            self._value = value.upper()
+
+
+    def __repr__(self):
+        return f'<WB Proj Num "{self._value}">'
+
+    def __str__(self):
+        return self._value
+
+    def __eq__(self, other):
+        return self._value == other._value
+
+    def __bool__(self):
+        return self._value is not None
+
+
+    @classmethod
+    def validate(cls, value):
+
+        # Return None as-is
+        if value is None:
+            return None
+
+        # Validate value field of WormbaseProjectNumber class
+        elif isinstance(value, WormbaseProjectNumber):
+            return WormbaseProjectNumber.validate(value._value)
+
+        # Check string value against RegEx
+        return isinstance(value, str) and bool(re.match(r'^PRJNA[0-9]+$', value, flags=re.IGNORECASE))
+
+
+# Test code
```

Comment:
@vlagrassa Excellent candidate for a unit test! Take a look at https://docs.python.org/3/library/unittest.html and consider writing a unit test file with these cases. 

Note: look into python unittest discover (it can find your test cases and run them).

Reply:
The unity test should be a different PR, no need to submit it for this release version.